comp/otelcol/otlp/components/processor/infraattributesprocessor/factory.go

```diff
@@ -19,13 +19,15 @@ import (
 var processorCapabilities = consumer.Capabilities{MutatesData: true}
 
 type factory struct {
-	tagger taggerClient
+	tagger     taggerClient
+	generateID GenerateKubeMetadataEntityID
```

Comment:
I assume this is needed because `comp/core/workloadmeta/collectors/util` is not a module?

Reply:
correct, workaround for now to avoid creating half a dozen other modules

Reply:
understood; Dinesh suggested this approach but if you want me to get back to the other dependencies first happy to do that

Reply:
I'm OK with the current approach, it's an incremental step. Can you leave a TODO here to remove generateID as a dependency in the future?

Reply:
I would recommend to take this incremental step as priority of OCB is unclear at the moment.